src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -759,13 +760,58 @@ void loop_inst::concatenated_memory_mapping::slice_mem(const int64_t num_iterati
         auto mem = sliced_mems[i];
         pointers_to_data[stride > 0 ? i : (num_iters - i - 1)] = reinterpret_cast<char*>(mem->lock(stream));
     }
+
     char* concat_data = reinterpret_cast<char*>(concatenated_mem->lock(stream, cldnn::mem_lock_type::read));
-    ov::reference::split(concat_data, concat_mem_shape, elem_size, axis, num_iters, pointers_to_data.data());
+
+    auto concate_layout = concatenated_mem->get_layout();
+    auto trait = format::traits(concate_layout.format);
+    if (format::is_blocked(concate_layout.format) || concate_layout.data_padding) {
```

Comment:
could you add comment here ? something like...
`BE CAREFUL: ov::reference::split is extremely slow. If we encounter any case where this code path is executed, we need to optimize it`

Reply:
OK, it is added.

Reply:
nit: I cannot see it. Am I missing something?

Reply:
I overwrited prevous commit while fixing windows build warning. Sorry I will update.